main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/input"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+)
+
+// Config ...
+type Config struct {
+	IpaPath           string          `env:"ipa_path"`
+	PkgPath           string          `env:"pkg_path"`
+	ItunesConnectUser stepconf.Secret `env:"itunescon_user,required"`
+	Password          stepconf.Secret `env:"password"`
+	AppPassword       stepconf.Secret `env:"app_password"`
+}
+
+func main() {
+	var cfg Config
+	if err := stepconf.Parse(&cfg); err != nil {
+		log.Errorf("Error: %s\n", err)
+		os.Exit(1)
+	}
+
+	stepconf.Print(cfg)
+	fmt.Println()
+
+	if err := cfg.validateEnvs(); err != nil {
+		failf("input error: %s", err)
+	}
+
+	password := string(cfg.Password)
+	if string(cfg.AppPassword) != "" {
+		password = string(cfg.AppPassword)
+	}
+
+	filePth := cfg.IpaPath
+	if cfg.PkgPath != "" {
+		filePth = cfg.PkgPath
+	}
+
+	cmd := command.New(`/Applications/Xcode.app/Contents/Applications/Application Loader.app/Contents/Frameworks/ITunesSoftwareService.framework/Support/altool`, "--upload-app", "-f", filePth, "-u", string(cfg.ItunesConnectUser), "-p", password)
+	cmd.SetStdout(os.Stdout)
+	cmd.SetStderr(os.Stderr)
+
+	fileName := filepath.Base(filePth)
+	log.Infof("Uploading - %s ...", fileName)
+	log.Printf("$ %s", cmd.PrintableCommandArgs())
+
+	if err := cmd.Run(); err != nil {
+		failf("Uploading IPA failed: %s", err)
+	}
+
+	fmt.Println()
+	log.Donef("IPA uploaded")
+}
+
+func (cfg Config) validateEnvs() error {
+	if err := input.ValidateIfNotEmpty(cfg.IpaPath); err != nil {
+		if err := input.ValidateIfNotEmpty(cfg.PkgPath); err != nil {
+			return fmt.Errorf("Neither ipa_path nor pkg_path is provided")
```

Comment:
error should not start with capital letter, this would produce: `$ input error: Neither ipa_path nor pkg_path is provided`